src/API/src/analytics/analytics.service.ts

```diff
@@ -0,0 +1,127 @@
+};
+
+@Injectable()
+export class AnalyticsService {
+  private umamiAuthToken: string;
+
+  constructor(
+    private logger: Logger,
+    private http: HttpService,
+    @InjectRepository(Occurrence)
+    private occurrenceRepository: Repository<Occurrence>,
+  ) {}
+
+  async init() {
+    this.umamiAuthToken = await getUmamiToken(this.http);
+  }
+
+  async eventAnalytics(
+    startAt: number,
+    endAt: number,
+    unit: string,
+    timezone: string,
+  ) {
+    try {
+      const events = await lastValueFrom(
+        this.http
+          .get(
+            `${process.env.ANALYTICS_API_URL}/websites/${process.env.NEXT_PUBLIC_ANALYTICS_ID}/events?start_at=${startAt}&end_at=${endAt}&unit=${unit}&tz=${timezone}`,
+            { headers: { Authorization: `Bearer ${this.umamiAuthToken}` } },
+          )
+          .pipe(
+            map((res: any) => {
+              return res.data;
+            }),
+          ),
+      );
+      let downloadFilteredButtonEvent = 0;
+      events.map((event: { x: string; y: number }) =>
+        event.x === 'download-filtered'
+          ? (downloadFilteredButtonEvent += event.y)
+          : '',
+      );
+      return downloadFilteredButtonEvent;
+    } catch (e) {
+      this.logger.error(e);
+      throw e;
+    }
+  }
+
+  async statsAnalytics(startAt: number, endAt: number) {
+    try {
+      const pageViews = await lastValueFrom(
+        this.http
+          .get(
+            `${process.env.ANALYTICS_API_URL}/websites/${process.env.NEXT_PUBLIC_ANALYTICS_ID}/stats?start_at=${startAt}&end_at=${endAt}`,
+            { headers: { Authorization: `Bearer ${this.umamiAuthToken}` } },
+          )
+          .pipe(
+            map((res: any) => {
+              return res.data;
+            }),
+          ),
+      );
+      return pageViews;
+    } catch (e) {
+      this.logger.error(e);
+      throw e;
+    }
+  }
+
+  async metricsAnalytics(startAt: number, endAt: number, type: string) {
+    try {
+      const metrics = await lastValueFrom(
+        this.http
+          .get(
+            `${process.env.ANALYTICS_API_URL}/websites/${process.env.NEXT_PUBLIC_ANALYTICS_ID}/metrics?start_at=${startAt}&end_at=${endAt}&type=${type}`,
+            { headers: { Authorization: `Bearer ${this.umamiAuthToken}` } },
+          )
+          .pipe(
+            map((res: any) => {
+              return res.data;
+            }),
+          ),
+      );
+      const numberCountries = metrics.length;
+      return numberCountries;
+    } catch (e) {
+      this.logger.error(e);
+      throw e;
+    }
+  }
+
+  async recordsAnalytics() {
+    try {
+      const approvedRecords = await this.occurrenceRepository
+        .createQueryBuilder('occurrence')
+        .leftJoin('occurrence.dataset', 'dataset')
+        .where('dataset.status = :status', { status: 'Approved' })
+        .getRawMany();
+      return approvedRecords.length;
+    } catch (e) {
+      this.logger.error(e);
+      throw e;
+    }
```

Comment:
Have we removed the bit where we total up the download_count?

Reply:
Yeah. When speaking with Ray he thought that may have caused confusion. So, I removed all the areas related to counting up the downloads. Now it's just number of vectors and individual download events (regardless of the number of rows downloaded)